optuna/integration/cma.py

```diff
@@ -0,0 +1,444 @@
+from __future__ import absolute_import
+
+import math
+import numpy
+import random
+
+import optuna
+from optuna import distributions
+from optuna.distributions import CategoricalDistribution
+from optuna.distributions import DiscreteUniformDistribution
+from optuna.distributions import IntUniformDistribution
+from optuna.distributions import LogUniformDistribution
+from optuna.distributions import UniformDistribution
+from optuna.samplers import BaseSampler
+from optuna.structs import StudyDirection
+from optuna.structs import TrialState
+from optuna import types
+
+try:
+    import cma
+    _available = True
+except ImportError as e:
+    _import_error = e
+    # CmaEsSampler is disabled because cma is not available.
+    _available = False
+
+if types.TYPE_CHECKING:
+    from typing import Any  # NOQA
+    from typing import Dict  # NOQA
+    from typing import List  # NOQA
+    from typing import Optional  # NOQA
+    from typing import Set  # NOQA
+
+    from optuna.distributions import BaseDistribution  # NOQA
+    from optuna.samplers.base import InTrialStudy  # NOQA
+    from optuna.structs import FrozenTrial  # NOQA
+
+MIN_SIGMA0 = 1e-10  # Minimum value of sigma0 to avoid ZeroDivisionError in cma.CMAEvotionStrategy.
+
+
+class CmaEsSampler(BaseSampler):
+    """A Sampler using cma library as the backend.
+
+    Note that parallel execution of trials may degrade the optimization performance of CMA-ES,
```

Comment:
This piece of information seems less important than the example. Would you mind putting this paragraph after the example?

Reply:
Thanks. I fixed it in https://github.com/pfnet/optuna/pull/447/commits/119372d476b904dc375f43115ab8b7aced6e6954.